src/app/services/auth.service.ts

```diff
@@ -63,6 +63,34 @@ export class AuthService {
       .toPromise()
       .then(data => data);
   }
+
+  changePassword(oldpass: string, pass1: string, pass2: string): Promise<any> {
+    return new Promise((resolve, reject) => {
+      if (pass1 !== pass2) {
+        reject(`New password doesn't match!`);
```

Comment:
Can you validate that directly in template or did you mean in a function in component? Yeah, we can update that later...